src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -386,16 +386,16 @@ private void validateAsset(AssetModel asset, Errors errors, SchemaValidationCont
   private void validateSchema(DatasetSpecificationModel schema, Errors errors) {
     SchemaValidationContext context = new SchemaValidationContext();
     List<TableModel> tables = schema.getTables();
-    if (tables != null && !tables.isEmpty()) {
+    if (tables == null || tables.isEmpty()) {
+      errors.rejectValue(
+          "schema", "IncompleteSchemaDefinition", "Dataset tables must be defined in the schema");
```

Comment:
It doesn't look like we have any unit tests for the validator...it might be a good opportunity to put some in

Reply:
It looks like its covered in DatasetRequestValidatorTest.java

Reply:
oh derp, you're right